test/scons/simple/SConstruct

```diff
@@ -1,5 +1,9 @@
 import os
 env = Environment(toolpath=[os.environ.get('EMSCRIPTEN_TOOL_PATH')])
 env.Tool('emscripten')
+
+# SCons does not inherit the environment
+env.AppendENVPath('PATH', os.environ.get('PATH'))
```

Comment:
IIRC adding the whole PATH here is should not be necessary.   i.e. we want to force all the users of the emscripten tool to do this in their downstream SConstruct I would hope

I was kind hoping that instead the emscripten tool could use the PATH to find the emcc root, rather than just injecting the entire path into the environment like this.   

How about using shutil.which in emscripten.py to find the emcc path: 
 https://docs.python.org/3/library/shutil.html#shutil.which
 
 e.g. `emscPath = os.path.dirname(shutil.which('emcc'))`

Reply:
I'm struggling to understand the first paragraph

We are technically not requiring the user to inject the whole PATH like I did in this test (for convenience), they could just add the path to Emscripten if they wanted to, e.g. `env.AppendENVPath('PATH', '/usr/lib/emscripten')`, and I think that is reasonable: in SCons, the default lex tool sets LEX to 'flex', the default yacc tool sets YACC to 'bison', the default CXX is 'g++'... So they are all assumed to be in the PATH.

Using the "external" PATH to look for the binary is possible but I think that is very weird behaviour 🤔

Reply:
IIRC with scons, that users PATH is normally ignored right?   what you call the external PATH.   So its not enough to have gcc in your PATH, it need to be in scons's own minimal idea of the PATH right?  something like `/usr/bin:/bin`.  

The problem for emcc is that emcc will almost never be in the limited PATH I think.. so we would like this Tool to work with whatever custom location where emcc lives, without them needing to spell this out in their SConstruct (I think). 

The problem I think is that we transition emsdk to use "find emcc in the PATH" as the way to find emcc, but scons by default won't do that.

I personally think `emscPath = os.path.dirname(shutil.which('emcc'))` seem reasonable, if a little unusual for scons, since scons doesn't normally like to honor the user's PATH.   scons is trying to be hermetic here, but i don't see any way to be hermetic while also recieving the path to emcc from the outside.

Receiving the emcc path via `EMSCRIPTEN_ROOT` seems just as "un-hermetic" as looking for it in the PATH, no?  Or am I miusunderstanding?

Reply:
If you don't want to look into the user's PATH then I think we can go back to requiring `EMSCRIPTEN_ROOT` and having `emscons` set that (like in your original PR).  

Then folks you don't use emscons will just have to set `EMSCRIPTEN_ROOT` themselves, but that doesn't seem so bad.

Reply:
Yes, SCons works like that. <https://www.scons.org/doc/production/HTML/scons-user.html#sect-execution-environments>

> If you don't want to look into the user's PATH then I think we can go back to requiring `EMSCRIPTEN_ROOT` and having `emscons` set that (like in your original PR).
> 
> Then folks you don't use emscons will just have to set `EMSCRIPTEN_ROOT` themselves, but that doesn't seem so bad.

If `EMSCRIPTEN_ROOT` is not set we can let e.g. CXX simply be 'em++' (like it is right now) and let it be searched in the (SCons) PATH when needed, maybe?

That looks like the least weird option to me because at least the environment variable involved, `EMSCRIPTEN_ROOT`, is not used for anything else. `PATH` on the other hand is something that everyone has set and for other reasons, so if it relies on that it seems to be more "magical" to me... At least if I was using a tool like this one I would never think that it could be searching for something in my "external" PATH, and not the canonical SCons one. But by the nature of the command `emscons scons` I would assume that `emscons` would be setting something like environment variables to `scons`, so I would not be surprised to know that something like `EMSCRIPTEN_ROOT` was being set.

Actually because of that I was thinking of changing the environment variable names that `emscons` uses to set the `pkg-config` parameters to be something specific for the tool to understand, like `EMSCONS_PKG_CONFIG_{LIBDIR,PATH}`. Then if the user used this tool without `emscons`, it would not look for the standard `PKG_CONFIG_{LIBDIR,PATH}` variables, that could be set for some reason, and use them (like it is doing now). Because that is not how SCons usually operates.

That's the way I think

Reply:
Sounds good to me.